File: src/main/java/me/wawwior/utils/hashing/hashes/DefaultHash.java
package me.wawwior.utils.hashing.hashes;

import me.wawwior.utils.hashing.Hash;

import java.security.MessageDigest;
import java.util.Arrays;

public class DefaultHash implements Hash {

    private final MessageDigest digest;

    DefaultHash(MessageDigest digest) {
        this.digest = digest;
    }

    @Override
    public byte[] hash(byte[] bytes) {
        return digest.digest(bytes);
    }

    @Override
    public boolean compare(byte[] bytes, byte[] hash) {
        if (hash.length != hashLength())
            return false;
        return Arrays.equals(hash(bytes), hash);
    }

    @Override
    public int hashLength() {
        return digest.getDigestLength();
    }
}

File: src/main/kotlin/me/wawwior/utils/hashing/hashes/DefaultHash.kt
package me.wawwior.utils.hashing.hashes

import me.wawwior.utils.hashing.Hash
import java.security.MessageDigest

class DefaultHash(private val messageDigest: MessageDigest) : Hash {

    override fun hashLength() = messageDigest.digestLength

    override fun hash(bytes: ByteArray): ByteArray = messageDigest.digest(bytes)

    override fun compare(bytes: ByteArray, hash: ByteArray): Boolean {
        if (hash.size != hashLength())
            return false
        return hash(bytes).contentEquals(hash)
    }
}